### ISSUE TITLE
Enable more stages for golden tests

### ISSUE THREAD

**Issue by theunrepentantgeek** [Azure/azure-service-operator#1853]
**Describe the current behavior**

For a number of reasons that were logical at the time, we are currently suppressing/excluding a large number of pipeline stages in `NewTestCodeGenerator()`. 

Revisting the list of suppressions in the context of PR #1851, both @matthchr and myself believe that many of these should now be enabled.

Doing so will create a lot of churn in the golden file output from the tests, requiring some close reading to spot anything untoward, so we want to make those changes in a separate PR.

**Comment by theunrepentantgeek** [comment]
Still required, but should be done at the same time as #2150.

**Comment by matthchr** [comment]
Some improvements have been made here but there's more to do. Might make sense to do after #2084

**Comment by matthchr** [comment]
May not actually need to do this -- what we need to do is move off JSON to Swagger

**Comment by theunrepentantgeek** [comment]
Retaining for now. @theunrepentantgeek to investigate whether we need these.

**Comment by theunrepentantgeek** [comment]
No longer required; we've decreased the number of suppressions organically and most of the remaining ones make sense.
